rust_src/src/dired_unix.rs

```diff
@@ -765,3 +765,19 @@ pub fn get_users() -> LispObject {
 
     list(&unames)
 }
+
+pub fn get_groups() -> LispObject {
+    let mut group_names = Vec::new();
+    let mut done = false;
+
```

Comment:
A more idiomatic approach might be:
```
loop {
    let gr = unsafe { getgrent() };
    if gr.is_null() {
        break;
    }
    group_names.push(unsafe { build_string((*gr).gr_name) });
}
```

Reply:
A more idiomatic Rust library would let you write:
```
while let Some(gr) = getgrent() {
    // use gr here
}
```

But libc exposes the C API mostly.

Reply:
Good catch, thank you. Looks much nicer.

Reply:
Do we need a wrapper for `getgrent` to apply @shaleh 's proposal ?